feat: add control slider for editing mod values

isidorosp opened the pull request:
A ControlSlider component has been added to allow for the easy modification of mod values, attributes (such as block), etc. The default example allows the selection of the block.
![image](https://user-images.githubusercontent.com/27592/103546280-05ddf480-4ea3-11eb-9a61-7cd8a6757a11.png)
![image](https://user-images.githubusercontent.com/27592/103546296-0aa2a880-4ea3-11eb-88da-766039d340cb.png)

isidorosp committed:
feat: add control slider for editing mod values

A ControlSlider component has been added to allow for the easy modification of mod values, attributes (such as block), etc. The default example allows the selection of the block.

adrianleb commented:
Very nice! being able to change the block is a great first use-case, the code looks good, could be done in a more scalable way but at this point that is early optimization and this is meant to be a style template after all.

The one ask I'd have if allowed to nitpick is having a few lines of documentation around the component / state to explain why that is there, and hint that people using the template can copy paste that patter and use it for mod values

isidorosp commented:
> Very nice! being able to change the block is a great first use-case, the code looks good, could be done in a more scalable way but at this point that is early optimization and this is meant to be a style template after all.
Happy to extend somehow if you had something in mind re: scalability. I just made it dumb enough so that it can work for anything we want it to right now since the idea is just to scaffold a template as fast as possible.

> The one ask I'd have if allowed to nitpick is having a few lines of documentation around the component / state to explain why that is there, and hint that people using the template can copy paste that patter and use it for mod values
Happy to add but can you be a little more specific about what kind of info you want where?

isidorosp committed:
feat: Add Color Control

Add Color Control Slider
Add some comments on instructions on how to use the control

isidorosp commented:
> Very nice! being able to change the block is a great first use-case, the code looks good, could be done in a more scalable way but at this point that is early optimization and this is meant to be a style template after all.
> 
> The one ask I'd have if allowed to nitpick is having a few lines of documentation around the component / state to explain why that is there, and hint that people using the template can copy paste that patter and use it for mod values

Added some comments and a color slider. Let me know if they're sufficient :)
